crates/bitwarden/src/vault/cipher/cipher.rs

```diff
@@ -343,11 +343,13 @@ fn build_subtitle_identity(first_name: Option<String>, last_name: Option<String>
 
 impl CipherView {
     pub fn generate_cipher_key(&mut self, key: &SymmetricCryptoKey) -> Result<()> {
-        let ciphers_key = Cipher::get_cipher_key(key, &self.key)?;
-        let key = ciphers_key.as_ref().unwrap_or(key);
+        let old_ciphers_key = Cipher::get_cipher_key(key, &self.key)?;
+        let old_key = old_ciphers_key.as_ref().unwrap_or(key);
 
         let new_key = SymmetricCryptoKey::generate(rand::thread_rng());
 
+        self.reencrypt_attachment_keys(old_key, &new_key)?;
```

Comment:
Could this be confusing for the mobile apps? Since we won't support saving older items without the attachments being updated first. Should we avoid generating a cipher key in that scenario.

Reply:
After discussion we should only update attachments keys if they use cipher keys here. If not it should still be the user key 🤯